types/material-ui/index.d.ts

```diff
@@ -1241,37 +1241,37 @@ declare namespace __MaterialUI {
         export interface IconMenuProps {
             // <Menu/> is the element that get the 'other' properties
             anchorOrigin?: propTypes.origin;
+            animated?: boolean;
             animation?: React.ComponentClass<Popover.PopoverAnimationProps>;
             className?: string;
-            iconButtonElement: React.ReactElement<IconButtonProps>;
+            clickCloseDelay?: number;
+            iconButtonElement: React.ReactElement<IconButton>;
             iconStyle?: React.CSSProperties;
             menuStyle?: React.CSSProperties;
+            onClick?(e: React.SyntheticEvent<{}>): void;
             onItemClick?(e: React.SyntheticEvent<{}>, item: MenuItem): void;
             onKeyboardFocus?(e: React.FocusEvent<{}>, isKeyboardFocused: boolean): void;
             onMouseDown?: React.MouseEventHandler<{}>;
             onMouseEnter?: React.MouseEventHandler<{}>;
             onMouseLeave?: React.MouseEventHandler<{}>;
             onMouseUp?: React.MouseEventHandler<{}>;
             onRequestChange?(opening: boolean, reason: string): void;
-            onClick?: React.MouseEventHandler<{}>;
             open?: boolean;
-            style?: React.CSSProperties;
             targetOrigin?: propTypes.origin;
-            clickCloseDelay?: number;
             useLayerForClickAway?: boolean;
 
-            animated?: boolean;
+            // Other properties from <Menu/>
             autoWidth?: boolean;
+            children?: Array<React.ReactElement<MenuItemProps>>;
```

Comment:
Just found the bug from my previous [comment](https://github.com/DefinitelyTyped/DefinitelyTyped/pull/22663/files#r165432206).

The problem is that I'm adding MenuItem conditionally. Example:
```
<MenuItem ... />
{this.props.someCondition &&
    <MenuItem ... />
}
<MenuItem ... />
```

So in my case here the type must be `Array<false|React.ReactElement<MenuItemProps>>`

Reply:
@andrioli
I found the bug in MenuItem's children, so fixed #22910 .
Your bug is fixed this PR, or not yet fixed?
(what means `Array<false>` ?)

Can you show the sample code which causes the error?

Reply:
No, is not yet fixed.

If you try to create a `IconMenu` with the items from my previous comment the compile will break.

Complete Example

```
import IconButton from 'material-ui/IconButton';
import IconMenu from 'material-ui/IconMenu';
import MenuItem from 'material-ui/MenuItem';
import PersonIcon from 'material-ui/svg-icons/social/person';
import * as React from 'react';

interface IconMenuBugProps {
  someCondition: boolean;
}

class IconMenuBug extends React.Component<IconMenuBugProps> {

  public render() {
    return (
      <IconMenu
        iconButtonElement={<IconButton><PersonIcon color="#AD3489" /></IconButton>}
      >
        <MenuItem value="0" primaryText="Hey" />
        {/* may be false */}
        {!this.props.someCondition &&
          <MenuItem value="1" primaryText="Ho" />
        }
        {/* may be false */}
        {this.props.someCondition &&
          <MenuItem value="2" primaryText="Let's Go" />
        }
      </IconMenu>
    );
  }

}
```

Note that the children array can also contains `false` values other than only `React.ReactElement<MenuItemProps>`

Reply:
Thank you, I understand.
I created #23387.

Sorry about any inconvenience.

Reply:
Thank you for the PR